checker_c#/N0t3b00kChecker/Db/CheckerDb.cs

```diff
@@ -0,0 +1,53 @@
+﻿using EnoCore.Checker;
+using Microsoft.Extensions.Logging;
+using MongoDB.Driver;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace N0t3b00kChecker.Db
+{
+    public class CheckerDb
+    {
+        private readonly ILogger<CheckerDb> logger;
+        private readonly IMongoCollection<NotebookUser> users;
+        private readonly InsertOneOptions insertOneOptions = new() { BypassDocumentValidation = false };
+
+        public CheckerDb(ILogger<CheckerDb> logger)
+        {
+            this.logger = logger;
+            this.logger.LogDebug($"CheckerDb() via {MongoConnection})");
+            var mongo = new MongoClient(MongoConnection);
+            var db = mongo.GetDatabase("N0t3b00kCheckerDatabase");
+            this.users = db.GetCollection<NotebookUser>("Users");
+            this.users.Indexes.CreateOne(new CreateIndexModel<NotebookUser>(Builders<NotebookUser>.IndexKeys
+                .Ascending(nu => nu.TaskChainId)));
+        }
+
+        public static string MongoHost => Environment.GetEnvironmentVariable("MONGO_HOST") ?? "localhost";

```

Comment:
Is that necessary for every checker? Shouldn't that be handled by the library?

Reply:
Yes, because checkers may want to set arbitrary indexes and use arbitrary database structures. In enowars4, gamemaster required an extra index over (roundId, serviceId) to select arbitrary users created during a certain round for a given team.

Reply:
Yeah, but that does not affect the environment variables or connection itself?

Reply:
Ah I thought you meant the class in general.

Do we want to specify mongo env vars across all checkerlibs? I assumed that since checker authors provide the checker's docker-compose.yml they will be able to supply matching values here.

Reply:
Not necessarily, but I would assume they should at least be the same for all dotnet checkers. Just feels like a bit too much boilerplate code that needs to be copy&pasted into each repository

Reply:
> Yes, because checkers may want to set arbitrary indexes and use arbitrary database structures. In enowars4, gamemaster required an extra index over (roundId, serviceId) to select arbitrary users created during a certain round for a given team.

Sounds like it would strongly go against the current service tenets.

Reply:
Against which one?

Reply:
Not tenets, but basically this one?
https://github.com/enowars/specification/blob/main/checker_protocol.md#relatedroundid

Reply:
Maybe strongly was worded too strongly :D

Reply:
The description there is very harsh and not according to our current tenets. I would agree that it describes the foo-proof way of doing things, but certainly not the best way.

Do we want to remove that line? The `taskChainId` clearly states that it should be used as a unique index, and if someone wants to (gracefully) recycle users from old rounds I'd encourage that.

Reply:
Maybe move the discussion to the Spec repo?